Target-MVD-Mathias/Views/Modal.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Modal.swift
+//  Target-MVD-Mathias
+//
+//  Created by Mathias Cabano on 10/6/17.
+//  Copyright © 2017 TopTier labs. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+protocol Modal {
+  var backgroundView: UIView! { get }
+  var dialogView: UIView! { get set }
+  
+  func show(animated: Bool)
+  func dismiss(animated: Bool)
+}
+
+extension Modal where Self: UIView {
+  
+  func show(animated: Bool) {
+    self.backgroundView.alpha = 0
+    UIApplication.shared.delegate?.window??.rootViewController?.view.addSubview(self)
+
+    self.dialogView.layoutIfNeeded()
+    self.backgroundView.layoutIfNeeded()
+    self.layoutIfNeeded()
+    
+    self.dialogView.center = CGPoint(x: self.center.x, y: self.frame.height + self.dialogView.frame.height / 2)
+    
+    if animated {
+      UIView.animate(withDuration: 0.35) {
+        self.backgroundView.alpha = 0.4
+      }
+      UIView.animate(withDuration: 0.35, delay: 0, usingSpringWithDamping: 0.7, initialSpringVelocity: 10, options: UIViewAnimationOptions(rawValue: 0), animations: {
+        self.dialogView.center = self.center
+      }, completion: nil)
+    } else {
+      self.backgroundView.alpha = 0.4
+      self.dialogView.center = self.center
+    }
+  }
+  
+  func dismiss(animated: Bool) {
+    if animated {
+      UIView.animate(withDuration: 0.35) {
+        self.backgroundView.alpha = 0
```

Comment:
Can't you add this inside of the next animation? I think they are executed at the same time and have the same duration after all.